### PR TITLE
Fixing CountOnes_Mixed() in bitvector

### PR DESCRIPTION
**Pull request by rodsan0**
also, added optimizations when compiling with SSE4.2

### CODE REVIEW

File: source/tools/BitVector.h [hunk 1]
```diff
@@ -19,6 +19,8 @@
 #define EMP_BIT_VECTOR_H
 
 #include <iostream>
+#include <nmmintrin.h> // testing smth out
```
Comment (mmore500) [inline source/tools/BitVector.h]: Will this work on systems without this header (non-intel systems?)?
also, what does the comment mean? If it's an old comment, we can remove it

**Comment by mmore500** [comment]
@mercere99 I put together this benchmark of bitcounting using stl bitset's count vs our manual bitcount without the popcnt instruction. Looks like there isn't any performance degradation in this case. (quickbench doesn't have error bars but says "equivalent" if you mouse over). Tested on latest Clang and GCC.

[http://quick-bench.com/niZXb401LvPdFQSLaTr08H0enrQ](http://quick-bench.com/jqDUbXiEOniMEsoYaA9-xtw5OdM#)

[http://quick-bench.com/p5vYxnoeO922MgY7UhFWXGAVimg](http://quick-bench.com/p5vYxnoeO922MgY7UhFWXGAVimg)

I think this was the last concern we wanted to address before pulling this in?

**Comment by mmore500** [comment]
In fact, if you inspect the highlighted assembly it looks equivalent (at a cursory glance)

**Comment by mercere99** [comment]
Excellent.  That's what it looked like, but it's good to know that's what's actually happening.  We could have a slightly faster bit-counter than what we've been using, but the advantage of using bitset makes that route well worth it.

**Pull request merged**